### PR TITLE
RAIL-3455 Attribute filtering bug fixing

### PR DESCRIPTION
**Pull request by plavemar**
- fixes for AttributeFilterButton component
- fixes for AttributeFilter component
- fixes for attribute parent-child filtering feature

JIRA: RAIL-3455

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by plavemar** [comment]
extended test

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 1]
```diff
@@ -162,8 +164,25 @@ const DropdownButton: React.FC<{
...
+
+        if (!element) {
+            return;
+        }
+
+        const roundedWidth = Math.ceil(element.getBoundingClientRect().width);
+        const displayItemCount = roundedWidth < element.scrollWidth;
+
+        setDisplayItemCount(displayItemCount);
+    });
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:184]: Should this have some dependencies?

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 2]
```diff
@@ -293,6 +337,7 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
     } = useCancelablePromise<IAttributeMetadataObject>(
         {
             promise: async () => {
+                console.log(currentFilter);
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:340]: Forgotten console.log, please remove it

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4301/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5911/ : FAILURE in 9m 23s

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 3]
```diff
@@ -530,18 +594,24 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
...
                 body={
                     <AttributeDropdownBody
                         items={elementsResult?.validOptions?.items ?? []}
-                        totalCount={elementsResult?.totalCount || LIMIT}
+                        totalCount={totalCount || LIMIT}
                         selectedItems={state.selectedFilterOptions}
                         isInverted={state.isInverted}
-                        isLoading={elementsStatus === "pending" || elementsStatus === "loading"}
+                        isLoading={
+                            elementsStatus === "pending" ||
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:610]: Detail: could be just `isElementsLoading() || isTotalCountLoading()`

File: libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx [hunk 4]
```diff
@@ -560,7 +630,14 @@ export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> =
     const { FilterError } = props;
 
     return elementsError || attributeError ? (
-        <FilterError error={elementsError?.message ?? attributeError?.message ?? "Unknown error"} />
+        <FilterError
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx:633]: There should be `|| totalCountError` in the condition too

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by plavemar** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1052/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4306/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1052/ : UNSTABLE in 19m 53s

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4308/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5915/ : SUCCESS in 21m 18s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1054/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4309/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1054/ : SUCCESS in 27m 00s

**Comment by plavemar** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1500

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4310/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/266/ : SUCCESS in 11m 00s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1500

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1235/ : SUCCESS in 1m 04s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1511/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/267/ : SUCCESS in 11m 05s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5920/ : SUCCESS in 16m 21s

**Pull request merged**